spec/profiles_spec.rb

```diff
@@ -12,18 +12,47 @@
   context 'with credentials' do
     let(:username) { 'username' }
     let(:password) { 'password' }
+    let(:phone_number) { '123-256-7890' }
+
+    describe 'get' do
+      before(:each) do
+        stub_upland_mobile_commons_request("profile?phone_number=#{phone_number}", method: :get).
+            to_return(status: 200, body: response_body)
+      end
+
+      context 'success' do
+        let(:response_body) { fixture('profiles/get.xml') }
+
+        it 'should return profile data' do
+          resp = subject.get(phone_number)
+          expect(resp).to_not be_nil
+          expect(resp['first_name']).to eq('John')
+          expect(resp['last_name']).to eq('Doe')
+          expect(resp['phone_number']).to eq('12025551234')
+        end
+      end
+
+      context 'error' do
+        let(:response_body) { fixture('error_profile_not_found.xml') }
+
+        it 'should return empty and not raise' do
+          resp = subject.get(phone_number)
+          expect(resp).to eq({})
```

Comment:
`nil` feels like a more normal response.

Reply:
I was torn on this one, but I agree returning `nil` for single result fetches feels more correct.